packages/pcd/semaphore-identity-v4/test/semaphore-identity-v4.spec.ts

```diff
@@ -79,7 +86,31 @@ describe("Semaphore Identity PCD", function () {
     expect(deserializedValid).to.eq(true);
     expect(deserialized.id).to.eq("2e02917f-99e7-4ee0-b2bb-0832d1e608e4-v4");
     expect(deserialized.claim.identity).to.deep.eq(
-      Identity.import("MjAzMTU5OTAwNzk4NDM2MTU4MTQ5MDY1Mjc0OTMxNDY=")
+      IdentityV4.import("MjAzMTU5OTAwNzk4NDM2MTU4MTQ5MDY1Mjc0OTMxNDY=")
+    );
+  });
+
+  it("v4PublicKey", function () {
+    const identity = new IdentityV4();
+    const privateKey = identity.export();
+    const pod = POD.sign({ a: { type: "int", value: 0n } }, privateKey);
+    expect(pod.signerPublicKey).to.eq(v4PublicKey(identity));
+    expect(encodePublicKey(identity.publicKey)).to.eq(v4PublicKey(identity));
+  });
+
+  it("v4PublicKeyToCommitment", function () {
+    const identity = new IdentityV4();
+    expect(identity.commitment.toString()).to.eq(
+      v4PublicKeyToCommitment(v4PublicKey(identity))
+    );
+  });
+
+  it("v3tov4Identity is deterministic", function () {
```

Comment:
This function also deserves a hard-coded input test to ensure it generates the same output.
Also a sanity check that the V4 identity's private key is 32 bytes.  Semaphore V4 allows any number of bytes, but PODs currently expect exactly 32.

Reply:
> It would be good to make sure we don't mess that up in this migration, since it's not actually signing any PODs with the resulting identity yet, but we intend to do so in future.

I do sign a POD with the v4 identity in this pr in `makeUpgradeUserWithV4CommitmentRequest`